staging/src/k8s.io/apimachinery/pkg/apis/config/validation/validation.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/apis/config"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateClientConnectionConfiguration ensures validation of the ClientConnectionConfiguration struct
+func ValidateClientConnectionConfiguration(cc *config.ClientConnectionConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if cc.QPS < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("qps"), cc.QPS, "must be non-negative"))
```

Comment:
this is not accurate... a negative number disables QPS rate limiting today

Reply:
this is the only validation I see around the client config struct today:

https://github.com/kubernetes/kubernetes/blob/113872798d34daa32277616b99a368244f1f7b65/pkg/proxy/apis/config/validation/validation.go#L190-L194

that could be useful to promote to a reusable method

Reply:
Thanks, I'll fix. 

For Burst, is there any range that's not accepted?